whylog/config/__init__.py

```diff
@@ -109,12 +109,24 @@ def load_config(cls, path):
             return YamlConfig(**config_paths), assistant_class
 
     @classmethod
-    def create_file_with_config_paths(cls, config_paths, path_to_config):
+    def _create_file_with_config_paths(cls, config_paths, path_to_config):
         with open(path_to_config, 'w') as config_file:
             config_file.write(yaml.safe_dump(config_paths, explicit_start=True))
         return path_to_config
 
 
+class ConfigFactorySelector(object):
+    SUPPORTED_TYPES = {'yaml': YamlConfigFactory}
+
+    @classmethod
+    def get_config(cls):
+        return cls.SUPPORTED_TYPES['yaml'].get_config()
```

Comment:
This function should find .whylog/config.yaml file and load the type of config (in this case it should be 'yaml') and then call get_config method with param - rest of the content of loaded file with specific paths and others initial params stored in the dict. Yaml files supports the dict structure.